### PR TITLE
Update organize-imports rule with a fix for givens

### PR DESCRIPTION
**Pull request by tgodzik**

### CODE REVIEW

**Review by ckipp01 — APPROVED** [review]
LGTM!

**Review by dos65 — COMMENTED** [review]
@tgodzik I took a look how to make it work with scala3 a did a [PR](https://github.com/tgodzik/metals/pull/12) on your branch

File: .scalafix3.conf [hunk 1]
```diff
@@ -4,6 +4,7 @@ rules = [
 
 OrganizeImports.groupedImports = Explode
 OrganizeImports.expandRelative = true
+OrganizeImports.removeUnused = false
```
Comment (ckipp01) [inline .scalafix3.conf:7]: This is because warn unused doesn't work with Scala 3 right?

Comment (dos65) [reply .scalafix3.conf:7]: yep + by default it's `true` that fails the internal validation for scala3 projects

**Review by dos65 — APPROVED** [review]

**Review by ckipp01 — APPROVED** [review]
LGTM!

File: metals/src/main/scala/scala/meta/internal/metals/ScalafixProvider.scala [hunk 2]
```diff
@@ -202,8 +202,22 @@ case class ScalafixProvider(
       .getOrElse("Unexpected error while running Scalafix.")
   }
 
-  private def scalafixConf: Option[Path] = {
+  private lazy val scala3DefaultConfig = {
```
Comment (tgodzik) [inline metals/src/main/scala/scala/meta/internal/metals/ScalafixProvider.scala:205]: Turns out that Scala organize imports will fail with an empty confix for Scala3, so I added a temporary config file.

**Pull request merged**